wsd/QuarantineUtil.cpp

```diff
@@ -78,6 +78,48 @@ namespace Quarantine
         return size;
     }
 
+    void deleteExpiredFiles(std::vector<std::string> &files)
+    {
+        if (!isQuarantineEnabled())
+            return;
+
+        // parameter is used in this function to avoid creating and sorting file list multiple times
+        if(files.empty())
+        {
+            Poco::File(LOOLWSD::QuarantinePath).list(files);
+            std::sort(files.begin(), files.end());
+        }
+
+        std::size_t timeLimit = LOOLWSD::getConfigValue<std::size_t>("quarantine_files.expiry_min", 30);
+        const auto timeNow = std::chrono::system_clock::now();
+        const auto ts = std::chrono::duration_cast<std::chrono::seconds>(timeNow.time_since_epoch()).count();
+
+        std::size_t index = 0;
+        while (!files.empty() && index < files.size())
+        {
+            FileUtil::Stat f(LOOLWSD::QuarantinePath + files[index]);
```

Comment:
In general it's better to do the syscall just once to stat the file to get its information; can we not factor out some 'bool deleteFile (name, timeNow, &totalSize); type function that we can scan do a single pass through the files with ? =)